docs/sql-pyspark-comparison-with-pandas.md

```diff
@@ -0,0 +1,401 @@
+
+
+### Load DataFrame
+
+To load DataFrame, there are APIs, usually, `spark.read.xxx` (or `spark.read.format("xxx")`)
+for PySpark and `pandas.read_xxx` for Pandas.
+
+```python
+# PySpark
+df = spark.read.csv("data.csv")
+```
+
+```python
+# Pandas
+df = pandas.read_csv("data.csv")
+```
+
+There are many sources available in both PySpark and Pandas. For example, CSV, JSON and
+Parquet are commonly supported but each supports different set of sources. There are pretty
+different set of options availabe as well, please see the API documentation for both sides.
+Note that, to match behaviours for PySpark to Pandas, `header=True` and `inferSchema=True`
+options are required.
+
+
+### Save DataFrame
+
+To load DataFrame, likewise, the APIs are `spark.read.xxx` (or `spark.read.format("xxx")`) for
+PySpark and `pandas_to_xxx` for Pandas.
+
+```
+df.to_csv("data.csv")
+```
+
+```
+df.write.csv("data.set")
+```
+
+Likewise, different set of sources are supported. Note that, to match behaviours for PySpark
+to Pandas, `header=True` option is required.
+
+
+### Inspect DataFrame
+
+DataFrame in both PySpark and Pandas can be expected in many ways. One of the way is to
+`describe()` as below:
+
+```
+# PySpark
+df.describe().show()
+```
+
+```
++-------+-------------+----+
+|summary|         city|rank|
++-------+-------------+----+
+|  count|            4|   4|
+|   mean|         null| NaN|
+| stddev|         null| NaN|
+|    min|      Chicago| 1.0|
+|    max|San Francisco| NaN|
++-------+-------------+----+
+```
+
+In Pandas, the same name function `describe()` can be called:
+
+```
+# Pandas
+df.describe()
+```
+
+```
+       rank
+count   3.0
+mean    2.0
+std     1.0
+min     1.0
+25%     1.5
+50%     2.0
+75%     2.5
+max     3.0
+```
+
+There are some differences, for instance, in case of PySpark, `NaN` is excluded but Pandas ignore `NaN` in some statistics. 
+
+To inspect columns, there are `dtype` API in PySpark and Pandas.
+
+```
+# PySpark
+df.dtypes
+```
+
+```
+[('city', 'string'), ('rank', 'double')]
+```
+
+```
+# Pandas
+df.dtypes
+```
+
+```
+0     object
+1    float64
+dtype: object
+```
+
+PySpark types are represented as Spark SQL types whereas they are NumPy types in case of Pandas.
+
+Note that, PySpark supports pretty printing of schema as below:
+
+```
+# PySpark
+df.printSchema()
+```
+
+```
+root
+ |-- city: string (nullable = true)
+ |-- rank: double (nullable = true)
+```
+
+
+### Interaction between PySpark and Pandas
+
+PySpark supports conversion to/from Pandas and Pandas UDFs out of the box.
+See [PySpark Usage Guide for Pandas with Apache Arrow](https://spark.apache.org/docs/latest/sql-pyspark-pandas-with-arrow.html).
+
+
+
+## Notable Differences
+
+There are other notable differences for many reasons, for instance,
+PySpark DataFrame's core is Spark SQL so some of APIs comply SQL, and this might be different
+comparing to Pandas context which somtimes causes some subtleties. See the differences below.
```

Comment:
comparing to Pandas -> compared to the Pandas
somtimes -> sometimes
causes some subtleties -> causes subtle differences?